changed message + removed message package

MaxandreOgeret opened the pull request.

Commit:
changed message + removed message package

Commit:
update readme

tambetm commented on src/h264_subscriber.cpp:
```diff
@@ -47,7 +48,8 @@ void H264Subscriber::internalCallback(const h264_image_transport_msgs::H264Packe
     }
-    convert_ctx = sws_getCachedContext(convert_ctx, picture->width, picture->height, AV_PIX_FMT_YUV420P, picture->width, picture->height, AV_PIX_FMT_BGR24, SWS_FAST_BILINEAR, NULL, NULL, NULL);
+    convert_ctx = sws_getCachedContext(convert_ctx, picture->width, picture->height, AV_PIX_FMT_YUV420P, picture->width, picture->height, AV_PIX_FMT_BGR24, SWS_FAST_BILINEAR,
+                                       nullptr, nullptr, NULL);
```
Why the last parameter is NULL, not nullptr?

MaxandreOgeret replied:
Indeed, no good reason, changed.

tambetm commented:
Comments about README:
* `git clone git@github.com:UT-ADL/h264_image_transport.git ./src` - this does not seem to be correct any more, now as you moved package to top level.
* `source ./devel/bash` -> `source devel/setup.bash`
* Link to `sekonix_camera_ut` should be updated to `nvidia_gmsl_driver_ros`.
* `http://wiki.ros.org/image_transport#republish` - there is no #republish anchor in the document, probably #Nodes or #Examples is the closest.

tambetm commented on src/h264_subscriber.cpp:
```diff
@@ -47,7 +48,8 @@ void H264Subscriber::internalCallback(const h264_image_transport_msgs::H264Packe
+                                       nullptr, nullptr, NULL);
     sensor_msgs::ImagePtr out = boost::make_shared<sensor_msgs::Image>();
     out->data.resize(picture->width * picture->height * 3);
     int stride = 3*picture->width;
```
One line below - isn't `uint8_t *dst = &out->data[0]` equivalent to `uint8_t *dst = out->data`? In that case you could skip creating `dst` entirely and use `put->data` directly in place of it.

MaxandreOgeret replied:
That parameter must be of type `uint8_t *const dst[]`, so pointer of pointer.
What we would need to do then is something like this : `&out->data.data()`. Alas it's not possible because `out->data.data()` yields a rvalue of type `unsigned char *`, and you cannot get a reference of that. You are condemned to store that value somewhere or find a better day to do things :laughing:

tambetm commented on src/h264_subscriber.cpp:
```diff
@@ -37,7 +38,7 @@ void H264Subscriber::subscribeImpl (ros::NodeHandle &nh, const std::string &base
+void H264Subscriber::internalCallback(const sensor_msgs::CompressedImage::ConstPtr& message, const Callback& user_cb)
 {
     avpkt.size = message->data.size();
     avpkt.data = (uint8_t*)&message->data[0];
```
Isn't `&message->data[0]` equivalent to `message->data`?

MaxandreOgeret replied:
This would be true for a C array. Here `message->data` is a std::vector.
But, in any case, to access a vector's underlying array,  [`std::vector<T,Allocator>::data`](https://en.cppreference.com/w/cpp/container/vector/data) should be used. 
Like so: `message->data.data()`

tambetm commented on src/h264_subscriber.cpp:
```diff
@@ -47,7 +48,8 @@ void H264Subscriber::internalCallback(const h264_image_transport_msgs::H264Packe
         ROS_ERROR("Could not decode H264 frame");
```
This should probably be warning, as it is non-fatal.

MaxandreOgeret replied:
IT should be removed altogether because an h264 frame != an image.

MaxandreOgeret replied:
Done

tambetm commented on package.xml:
```diff
@@ -0,0 +1,32 @@
+  <version>0.0.0</version>
+  <description>The h264_image_transport package</description>
+
+  <maintainer email="adl@ut.ee"> University of Tartu Autonomous Driving Lab</maintainer>
```
Add yourself as `<author>`.

MaxandreOgeret replied:
done

tambetm replied:
This doesn't seem to be done - forgot to commit?

MaxandreOgeret replied:
indeed, forgot to add the file before commit.

MaxandreOgeret replied:
done

tambetm commented on package.xml:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0"?>
+<package>
+  <name>h264_image_transport</name>
+  <version>0.0.0</version>
```
Bump to 0.0.1.

MaxandreOgeret replied:
done

tambetm replied:
This doesn't seem to be done - forgot to commit?

MaxandreOgeret replied:
done

tambetm reviewed (approved):
Mostly good to go, my comments are mainly suggestions.

Commit:
Cleaned, removed deprecated function calls, pr comments

Commit:
readme

MaxandreOgeret commented:
@tambetm 
>     * `git clone git@github.com:UT-ADL/h264_image_transport.git ./src` - this does not seem to be correct any more, now as you moved package to top level.

Indeed, changed.

>     * `source ./devel/bash` -> `source devel/setup.bash`

Done

>     * Link to `sekonix_camera_ut` should be updated to `nvidia_gmsl_driver_ros`.

Done

>     * `http://wiki.ros.org/image_transport#republish` - there is no #republish anchor in the document, probably #Nodes or #Examples is the closest.

Yes, the anchor has been removed. The ROS doc still has some links to that anchor. I changed  the link.

Commit:
added clang format+tidy files

tambetm commented on README.md:
````diff
@@ -19,12 +13,15 @@ This repo has been created to be used in pair with the [nvidia_gmsl_driver_ros](
 - Clone the repo
   ```bash
-  git clone git@github.com:UT-ADL/h264_image_transport.git ./src
+  mkdir src
````
Not necessary, already created above.

MaxandreOgeret replied:
Done

tambetm commented on README.md:
````diff
@@ -19,12 +13,15 @@ This repo has been created to be used in pair with the [nvidia_gmsl_driver_ros](
   ```bash
-  git clone git@github.com:UT-ADL/h264_image_transport.git ./src
+  mkdir src
+  cd src
````
I would suggest combining this with `cd catkin_ws` above - `cd catkin_ws/src`. Then it can be deleted.

MaxandreOgeret replied:
Done

tambetm commented on README.md:
````diff
@@ -19,12 +13,15 @@ This repo has been created to be used in pair with the [nvidia_gmsl_driver_ros](
   ```
 - Build and source the workspace
   ```bash
   catkin_make
````
`catkin build` is preferred.

MaxandreOgeret replied:
changed

tambetm reviewed (changes requested):
Just some minor changes still remain to be done.

Commit:
Improved readme + added author.

MaxandreOgeret commented:
@tambetm, I believe this PR has been made obsolete by https://github.com/UT-ADL/h264_image_transport/pull/2

MaxandreOgeret commented:
@tambetm, This PR has been made obsolete by https://github.com/UT-ADL/h264_image_transport/pull/2

tambetm commented:
Reopening this, I'd like to keep `h264_image_transport` and `gst_image_transport` separate.

tambetm commented:
This is good, should be merged once the [nvidia_gmsl_driver_ros](https://github.com/UT-ADL/nvidia_gmsl_driver_ros) is successfully installed and tested in the car.

tambetm committed:
Merge branch 'master' into feature/2022-08-16_ut_nvidia_drive_resize
